### PR TITLE
Adds HasPropWithValue Pickler

### PR DESCRIPTION
**Pull request by bp-kelley**
Adds the ability to pickle HasPropWithValue queries.

### CODE REVIEW

**Comment by bp-kelley** [comment]
@greglandrum I'm not sure if we need to bump the pickle version, I don't remember if you did that for the HasPropQuery and I forgot to check,

File: Code/GraphMol/MolPickler.cpp [hunk 1]
```diff
@@ -461,6 +463,13 @@ void pickleQuery(std::ostream &ss, const Query<int, T const *, true> *query) {
         const auto &pval = std::get<1>(v);
         streamWrite(ss, MolPickler::QUERY_VALUE, pval);
       } break;
+      case 6: {
+        auto v =
+            boost::get<std::tuple<MolPickler::Tags, Dict::Pair, double>>(qdetails);
+        streamWrite(ss, std::get<0>(v));
```
Comment (bp-kelley) [inline Code/GraphMol/MolPickler.cpp:469]: I should make a note that it's easier to pickle the double value (tolerance) first.

File: Code/GraphMol/MolPickler.cpp [hunk 2]
```diff
@@ -609,6 +618,48 @@ Query<int, T const *, true> *buildBaseQuery(std::istream &ss, T const *owner,
...
+          res = makePropQuery<T, bool>(pair.key, rdvalue_cast<bool>(pair.val), tolerance);
+          break;
+        case RDTypeTag::FloatTag:
+          res = makePropQuery<T, float>(pair.key, rdvalue_cast<float>(pair.val), tolerance);
+          break;
+        case RDTypeTag::DoubleTag:
+          res = makePropQuery<T, double>(pair.key, rdvalue_cast<double>(pair.val), tolerance);
+          break;
+        case RDTypeTag::StringTag:
+          // what about bit vect?if (pair.key != "HasPropValueQuery")
```
Comment (bp-kelley) [inline Code/GraphMol/MolPickler.cpp:649]: Remove comment

File: Code/GraphMol/SubstructLibrary/catch_tests.cpp [hunk 3]
```diff
@@ -326,4 +326,4 @@ TEST_CASE("ExtendedQueryMol") {
     CHECK(!ssslib.hasMatch(xqm));
   }
 }
-#endif
\ No newline at end of file
+#endif
```
Comment (bp-kelley) [inline Code/GraphMol/SubstructLibrary/catch_tests.cpp:329]: I shouldn't have touched this file...  I'll double check

**Comment by greglandrum** [comment]
> @greglandrum I'm not sure if we need to bump the pickle version, I don't remember if you did that for the HasPropQuery and I forgot to check,

I bumped it in #7680. Since we haven't done a release since then, I don't think we need to do it again.

**Review by greglandrum — CHANGES_REQUESTED** [review]
Most of this looks good, but the tests aren't particularly comprehensive.
I did a PR against your branch with more tests

**Comment by bp-kelley** [comment]
@greglandrum nice!

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**